bdk_electrum_example/src/electrum.rs

```diff
@@ -0,0 +1,205 @@
+    }
+
+    /// Scan for a list of scripts, and create an initial [`ChainGraph`] candidate update.
+    /// This update will only contain [`Txid`]s in SparseChain, and no actual transaction data.
+    ///
+    /// User needs to fetch the required transaction data and update the [`ChainGraph`] before applying it.
+    pub fn spk_txid_scan(
+        &self,
+        spks: impl Iterator<Item = Script>,
+        local_chain: &BTreeMap<u32, BlockHash>,
+    ) -> Result<SparseChain, ElectrumError> {
+        let mut dummy_keychains = BTreeMap::new();
+        dummy_keychains.insert((), spks.enumerate().map(|(i, spk)| (i as u32, spk)));
+
+        Ok(self.wallet_txid_scan(dummy_keychains, None, local_chain)?.0)
+    }
+
+    /// Scan for a keychain tracker, and create an initial [`KeychainScan`] candidate update.
+    /// This update will only contain [`Txid`]s in SparseChain, and no actual transaction data.
+    ///
+    /// User needs to fetch the required transaction data and update the [`KeychainScan`] before applying it.
+    pub fn wallet_txid_scan<K: Ord + Clone>(
+        &self,
+        scripts: BTreeMap<K, impl Iterator<Item = (u32, Script)>>,
+        stop_gap: Option<usize>,
+        local_chain: &BTreeMap<u32, BlockHash>,
+    ) -> Result<(SparseChain, BTreeMap<K, u32>), ElectrumError> {
+        let mut sparse_chain = SparseChain::default();
+
+        // Check for reorgs.
+        // In case of reorg, new checkpoints until the last common checkpoint is added to the structure
+        for (&existing_height, &existing_hash) in local_chain.iter().rev() {
+            let current_hash = self
+                .client
+                .block_header(existing_height as usize)?
+                .block_hash();
+            sparse_chain
+                .insert_checkpoint(BlockId {
+                    height: existing_height,
+                    hash: current_hash,
+                })
+                .expect("This never errors because we are working with a fresh chain");
+
+            if current_hash == existing_hash {
+                break;
+            }
+        }
+
+        // Insert the new tip
+        let (tip_height, tip_hash) = self.get_tip()?;
+        if sparse_chain
+            .insert_checkpoint(BlockId {
+                height: tip_height,
+                hash: tip_hash,
+            }).is_err() {
+                // This means our existing chain tip has been reorged out.
+                return Err(ElectrumError::Reorg(tip_height));
+            }
+
+
+        let mut keychain_index_update = BTreeMap::new();
+
+        // Fetch Keychain's last_active_index and all related txids.
+        // Add them into the KeyChainScan
+        for (keychain, mut scripts) in scripts {
+            let mut last_active_index = 0;
+            let mut unused_script_count = 0usize;
+
+            while let Some((index, script)) = scripts.next() {
+                let history = self
+                    .script_get_history(&script)?
```

Comment:
Lets use `batch_script_get_history` here.

Reply:
I didn't do this one because of the comment here https://github.com/LLFourn/bdk_core_staging/pull/77#discussion_r1050440828 and also couldn't figure how to use `stop_gap` in that case. Let me know if you intend to fetch for 2^32 scripts anyway and should we remove stop gap?? 

The other way is by calling `script_pubkeys_by_keychain` which will give us iterator over only derived scripts.. Let me know if that's the way we wanna go here..

Reply:
Sorry I responded in the linked convo.